src/common.rs

```diff
@@ -789,6 +805,54 @@ fn unescape_string_url(input: &wstr) -> Option<WString> {
     Some(str2wcstring(&result))
 }
 
+/// Un-escapes a PCRE2-escaped string literal.
+///
+/// This reverses the mapping of [`escape_string_pcre2()`] but also unescapes additional classes of
+/// characters that `escape_string_pcre2()` does not escape but the spec _permits_ the escaping of,
+/// such as a literal \n (the literal backslash charecter followed by the letter `n`).
+///
+/// Note that not all possible PCRE2 escapes are currently supported!
+///
+/// Does not assume ASCII input. Assumes all input is valid PCRE2 and was formed strictly by
+/// escaping literal strings in the first place (i.e. assume no functional regex operators like
+/// `(..)`, `|`, or `[..]`, and only change escapes like `\.` or `\\` into `.` or `|`, respectively.
+/// Similarly, assume that we do not have backslash-denoted character classes (i.e. `\s` or `\w`)
+/// or a backreference (in the case of something like `\1`), i.e. that `\` is *always* an escape.
+///
+/// PCRE2 supports \Q..\E to mean "insert everything between \Q and \E exactly as it appears", but
+/// we explictly don't support that here.
+fn unescape_string_pcre2(input: &wstr) -> Option<WString> {
+    // Output will be strictly less than or equal to input in length
+    let mut result = WString::with_capacity(input.len());
+    let mut chars = input.chars();
+    while let Some(c) = chars.next() {
+        match c {
+            // It is invalid for the input to contain a trailing, unpaired `\\`
+            '\\' => result.push(match chars.next()? {
+                'a' => '\x07',                               // alarm
+                'c' => (chars.next()? as u8 & 0x1F) as char, // control codes
+                'e' => '\x1b',                               // escape
+                'f' => '\x0c',                               // form feed
+                'n' => '\n',
+                'r' => '\r',
+                't' => '\t',
+                // TODO: Consider implementing support for the following, noting that the likelihood
```

Comment:
`string escape` should do the right thing. If we add what is described by the comments it will be a breaking change.